src/main/kotlin/repository/GenericRepository.kt

```diff
@@ -0,0 +1,24 @@
+package repository
+
+import java.util.*
+
+open class GenericRepository<Item> : Repository<Item> where Item : UniqueItem {
+    private val items: MutableList<Item> = mutableListOf()
+
+    override fun getItem(id: UUID): Item? =
```

Comment:
It's better to name get methods like this: "findItem", like in Spring. Kotlin moves away from the concept of naming get methods (on objects expecially): "getId, getName etc.". Instead we use "id, name etc."

Minor